### PR TITLE
[#51] Set up iOS distribution

### LINKED ISSUE #51: [Chore] Set up iOS deployment with Github Actions
## Why

App must be able to build and release iOS app to showcase developer can setup pipeline
 
## Acceptance Criteria

- [ ] Enable iOS workflows in repository
- [ ] Setup workflow to deploy iOS app to Firebase
- [ ] Setup workflow to deploy iOS app to TestFlight

### PR DESCRIPTION
**Pull request by lydiasama**
- Closes https://github.com/nimblehq/flutter-ic-lydia-ryan/issues/51

## What happened 👀

- Resolved CI/CD scripts
- Resolved incompatible dependencies
- Now we can distribute iOS app by this following environment:
  - Staging 👉🏻  **Testflight**
  - Production 👉🏻  **Testflight** and **Firebase App Distribution** 🎉 

## Insight 📝

- We did wrong setup `TEAM_ID`
- We forgot to add `APP_STORE_KEY_ID` and `APP_STORE_ISSUER_ID` in `Constants.rb`
- We need to add the correct Profiling and Certification for all build flavors:
  
  SCR-25660721-knrh

- We need to add `\n` (Enter a new line) before and end of the `APPSTORE_CONNECT_API_KEY` content in Github Secrets as the fastlane cannot get the key. Ref [here](https://github.com/fastlane/fastlane/issues/20593#issuecomment-1235531580)
- At the script, removed `environment: staging` since we did not save the Githubs Secrets in that environment. Thus, the `staging` environment has been removed.

## Proof Of Work 📹

**Testflight Staging:**

Screenshot 2566-07-21 at 11 43 51 AM

**Testflight Production:**

Screenshot 2566-07-21 at 11 44 11 AM

**Firebase Production:**

Screenshot 2566-07-21 at 11 42 43 AM

### CODE REVIEW

**Review by ryan-conway — APPROVED** [review]
🥳

File: ios/fastlane/Constants/Constants.rb [hunk 1]
```diff
@@ -86,6 +86,16 @@ def self.MAIN_TARGET_NAME
   end
 
   def self.DEV_PORTAL_TEAM_ID
-      '4TWS7E2EPE'
-    end
+    '4TWS7E2EPE'
+  end
+
+  # App Store Connect API Key ID
+  def self.APP_STORE_KEY_ID
```
Comment (markgravity) [inline ios/fastlane/Constants/Constants.rb]: What do think about moving these values to the `Environments.rb`? 🙏

**Review by manh-t — APPROVED** [review]
good job 👏

**Comment by lydiasama** [comment]
@ryan-conway I'm still figuring it out. 🙇🏻‍♀️  Will keep you updated. 🙏🏻

**Review by markgravity — APPROVED** [review]

**Pull request merged**